crates/rari-doc/src/templ/templs/links/cssxref.rs

```diff
@@ -67,11 +84,38 @@ pub fn cssxref_internal(
         _ => slug,
     };
 
-    let url = format!(
-        "/{}/docs/Web/CSS/{slug}{}",
-        locale.as_url_str(),
-        anchor.unwrap_or_default()
-    );
+    let base_url = format!("/{}/docs/Web/CSS/", locale.as_url_str());
+    // Determine the URL path based on the new structure
+    let mut url_path = if name.starts_with("&lt;") || name.starts_with('<') {
+        // Types go under Web/CSS/Reference/Values
+        format!("Reference/Values/{slug}")
+    } else if name.starts_with(':') {
+        // Pseudo-classes and pseudo-elements go under Web/CSS/Reference/Selectors
+        format!("Reference/Selectors/{slug}")
+    } else if name.starts_with('@') {
+        // At-rules go under Web/CSS/Reference/At-rules
+        format!("Reference/At-rules/{slug}")
+    } else if name.ends_with("()") {
+        // Functions go under Web/CSS/Reference/Values
+        format!("Reference/Values/{slug}")
+    } else {
+        // Everything else: check Properties first
+        let url_path = format!("Reference/Properties/{slug}");
+        let url = format!("{}{}", &base_url, &url_path);
+        if RariApi::get_page_nowarn(&url).is_ok() {
+            url_path
+        } else {
+            // Fallback to Values
+            format!("Reference/Values/{slug}")
+        }
+    };
+
+    if RariApi::get_page_nowarn(&format!("{}{}", &base_url, &url_path)).is_err() {
```

Comment:
I don't think we should be silently falling back to redirect resolution: it's just hiding the logic of this within a massive map, while pretending the resolution is deterministic.

For instance, the only reason `CSSxRef("<url>")` resolves to [Web/CSS/Reference/Values/url_value](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Values/url_value) is because [Web/CSS/url](https://developer.mozilla.org/en-US/docs/Web/CSS/url) redirects there - this feels bad, and it doesn't look like anything will warn us of that.

At the very least we should get some kind of warning so we can add cases like these to the "special case mappings" `match` above, or simply move the content.